File: Add Binary.py
class Solution:
	# @param a, a string
	# @param b, a string
	# @return a string
	def addBinary(self, a, b):
		index = 0
		carry = 0
		sum = ""
		while (carry > 0 or index < len(a) or index < len(b)):
			digit = carry
			if (index < len(a) and a[len(a)-1-index] == '1'):
				digit = digit + 1
			if (index < len(b) and b[len(b)-1-index] == '1'):
				digit = digit + 1
			carry = digit // 2
			digit = digit % 2
			if (digit == 1):
				sum = "1" + sum
			else:
				sum = "0" + sum
			index = index + 1
		return sum

File: Add Binary.java
public class Solution {
    public String addBinary(String a, String b) {
        String ret = "";
        int carry = 0;
        int index = 0;
        while (carry > 0 || index < a.length() || index < b.length()) {
            int digit = carry;
            if (index < a.length() && a.charAt(a.length()-1-index) == '1')
                digit++;
            if (index < b.length() && b.charAt(b.length()-1-index) == '1')
                digit++;
            carry = digit / 2;
            digit = digit % 2;
            ret = (digit==1?"1":"0") + ret;
            index++;
        }
        if (ret.isEmpty())
            return "0";
        return ret;
    }
}
